src/test/integration/cpFromPod.ts

```diff
@@ -0,0 +1,78 @@
+import assert from 'node:assert';
+import fs from 'node:fs';
+import path from 'node:path';
+import os from 'node:os';
+import { CoreV1Api, KubeConfig, V1Pod } from '../../index.js';
+import { Cp } from '../../cp.js';
+import { generateName } from './name.js';
+
+export default async function cpFromPod() {
+    const kc = new KubeConfig();
+    kc.loadFromDefault();
+
+    const coreV1Client = kc.makeApiClient(CoreV1Api);
+    const cp = new Cp(kc);
+
+    const testPodName = generateName('cp-test-pod');
+    const namespace = 'default';
+
+    const pod = new V1Pod();
+    pod.metadata = { name: testPodName };
+    pod.spec = {
+        containers: [
+            {
+                name: 'test-container',
+                image: 'busybox',
+                command: ['sh', '-c', 'echo "test content" > /tmp/test.txt && sleep 3600'],
+            },
+        ],
+        restartPolicy: 'Never',
+    };
+
+    console.log(`Creating pod ${testPodName}`);
+    await coreV1Client.createNamespacedPod({ namespace, body: pod });
+
+    console.log('Waiting for pod to be ready...');
+    let podReady = false;
+    for (let i = 0; i < 30; i++) {
+        const currentPod = await coreV1Client.readNamespacedPod({ name: testPodName, namespace });
+        if (currentPod.status?.phase === 'Running') {
+            podReady = true;
+            break;
+        }
+        await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
Here and on line 64, can you use `setTimeout()` from `node:timers/promises`. Then, you can `await setTimeout()` directly without needing to create an additional `Promise`.